# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/Scope.cpp
# Commit: f1bb16079c0e
# Full Hash: f1bb16079c0e4aebb2096698dd24ae0001d650bc
# Author: Jan de Mooij <jdemooij@mozilla.com>
# Date: 2021-03-06 09:33:16
# Regressor Bug: 1689413
# File Overlap Count: 1
# Description:
#   Bug 1689413 part 4 - Move ObjectFlags from BaseShape to Shape. r=jonco
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D106974
# ==============================================================================

diff -r 211c4f80a286 -r f1bb16079c0e js/src/vm/Scope.cpp
--- a/js/src/vm/Scope.cpp	Fri Mar 05 19:10:05 2021 +0000
+++ b/js/src/vm/Scope.cpp	Fri Mar 05 19:10:06 2021 +0000
@@ -114,7 +114,8 @@
   }
 
   jsid id = NameToId(name->asPropertyName());
-  Rooted<StackShape> child(cx, StackShape(shape->base(), id, slot, attrs));
+  Rooted<StackShape> child(
+      cx, StackShape(shape->base(), shape->objectFlags(), id, slot, attrs));
   return cx->zone()->propertyTree().getChild(cx, shape, child);
 }
 